### ISSUE TITLE
Tensorboard incorrect logging with DataParallel training 

### ISSUE THREAD

**Issue by piseabhijeet** [PyTorchLightning/pytorch-lightning#9839]
## 🐛 Bug

Tensorboard logging metrics from different GPUs when using DataParallel training. 

### To Reproduce

Use any arbitrary toy model with DP as an accelerator and in **training_step()**, include the below line of code:

`self.log(mode + '_loss', sum(losses), on_epoch=False, on_step=True, prog_bar=True, logger=True, sync_dist=True)`

In the trainer, specify tensorboard logger with each iteration logging 
`pl.Trainer(...., log_every_n_steps=1))`

-->

### Expected behavior

According to the docs, for logging and DataParallel training, when we use sync_dist=True flag, it should log the metrics from the rank 0 worker/ main GPU, however, i observe a discrepancy across the logs:
At one step, metrics from GPU 0 is logged and at other step metric 1 is logged:

This behavior is completely random and it logs from any GPU, for instance, see the following:
![image](https://user-images.githubusercontent.com/57705684/136214471-e56bf6eb-c04c-4cdc-a71c-7284f98cd0a4.png)
here, the logging happens from GPU 0.

and this:
![image](https://user-images.githubusercontent.com/57705684/136214663-27610373-b1e8-4cf8-8a4a-d440a3d66f03.png)
here, the logging happens from GPU 1.

is sync_dist=True not doing its job or am i missing something like the reducer node can be any of the GPUs and the logging can happen from any node?

### Environment

- PyTorch Lightning Version: 1.4.8
- PyTorch Version : 1.8.1+cu111
- Python version: 3.8.5
- OS (e.g., Linux): linux
